apps/ios/SimpleXChat/ChatTypes.swift

```diff
@@ -1498,6 +1507,44 @@ public enum SndGroupEvent: Decodable {
     }
 }
 
+public enum RcvConnEvent: Decodable {
+    case switchQueue(phase: SwitchPhase)
+    
+    var text: String {
+        switch self {
+        case let .switchQueue(phase):
+            if case .completed = phase {
+                return NSLocalizedString("changed address for you", comment: "chat item text")
+            }
+            return NSLocalizedString("changing address...", comment: "chat item text")
+        }
+    }
+}
+
+public enum SndConnEvent: Decodable {
+    case switchQueue(phase: SwitchPhase, member: GroupMemberRef?)
+    
+    var text: String {
+        switch self {
+        case let .switchQueue(phase, member):
+            var forMember = ""
+            if let m = member {
+                forMember = String.localizedStringWithFormat(NSLocalizedString(" for %@", comment: "for member name"), m.profile.profileViewName)
+            }
+            if case .completed = phase {
+                return String.localizedStringWithFormat(NSLocalizedString("you changed address%@", comment: "chat item text"), forMember)
+            }
+            return String.localizedStringWithFormat(NSLocalizedString("changing address%@...", comment: "chat item text"), forMember)
+        }
```

Comment:
again this implies some similar structure in different languages, or makes it harder to translate out of context.. it was ok in terminal where it's only in English. Just make it separate texts depending on whether member is present or not?

Reply:
these text combinations in scope of a single message save a bit of code but then it may cause more problems than they're worth.. even for English only this is harder to understand than full messages with a single interpolation